libass/src/library.rs

```diff
@@ -29,20 +29,18 @@ pub struct Library<'a> {
 }
 
 impl<'a> Library<'a> {
-    pub fn new() -> Option<Self> {
+    pub fn new() -> Result<Self> {
         let lib = unsafe { ffi::ass_library_init() };
-        if lib.is_null() {
-            None
-        } else {
-            Some(Library {
-                handle: unsafe { NonNull::new_unchecked(lib) },
-                phantom: PhantomData,
-            })
-        }
+        err_if_null!(lib);
+        Ok(Library {
+            handle: unsafe { NonNull::new_unchecked(lib) },
+            phantom: PhantomData,
+        })
     }
 
-    pub fn set_fonts_dir(&mut self, fonts_dir: &CStr) {
-        unsafe { ffi::ass_set_fonts_dir(self.handle.as_ptr(), fonts_dir.as_ptr()) }
+    pub fn set_fonts_dir(&mut self, fonts_dir: &str) {
+        let fonts_dir = CString::new(fonts_dir).unwrap();
+        unsafe { ffi::ass_set_fonts_dir(self.handle.as_ptr(), fonts_dir.as_ptr() as *const i8) }
```

Comment:
```suggestion
        unsafe { ffi::ass_set_fonts_dir(self.handle.as_ptr(), fonts_dir.as_ptr()) }
```
The cast is not needed anymore, and it would cause compilation on platforms which use an unsigned char (like ARM) to fail.